src/server/acl/acl_family.cc

```diff
@@ -157,12 +163,31 @@ std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(CmdArgList args) {
 
 }  // namespace
 
+void AclFamily::StreamUpdatesToAllProactorConnections(std::string_view user, uint32_t update_cat) {
+  auto update_cb = [user, update_cat]([[maybe_unused]] size_t id, util::Connection* conn) {
+    DCHECK(conn);
+    auto connection = static_cast<facade::Connection*>(conn);
+    auto ctx = static_cast<ConnectionContext*>(connection->cntx());
+    if (ctx && user == ctx->authed_username) {
+      ctx->acl_categories = update_cat;
+    }
+  };
+
```

Comment:
SetUser is called so rarely that we can neglect the cost of traversing all connections? (of which there are also just a few usually?)

Reply:
Damn... What if the context is performing squashing? 😭 

We can copy the value (instead of referencing the parent), but then how do we propagate changes? On each batch? 😕

Reply:
If you want you can drop support for squashing and I will think how to enable support for your features

Reply:
Btw, because Redis is single-threaded it's impossible for ACL rules to change while its executing a script / a multi transaction, right?

Reply:
> SetUser is called so rarely that we can neglect the cost of traversing all connections? (of which there are also just a few usually?)

Yes that's true but it doesn't hurt to not traverse if the user was just created :)

>Damn... What if the context is performing squashing? 😭 We can copy the value (instead of referencing the parent), but then how do we propagate changes? On each batch? 😕 

I already thought of this :man_cook: ! It won't affect us. Why? Because, sure we update the "stub" contexts as well BUT validation is done on the `parent` and not on the `stub` context. Therefore, we don't really care, since the update done to stub contexts won't affect anything

>If you want you can drop support for squashing and I will think how to enable support for your features

It's all fine and if it wasn't we would fix it :rocket: No rush here

> Btw, because Redis is single-threaded it's impossible for ACL rules to change while its executing a script / a multi transaction, right?

Redis is single threaded indeed but connections can be multiplexed just like on a single thread you can run multiple tasks interleaved with each other (to the user this looks like multihreading, in reality it's just a single thread that interleaves task execution). That being said, this example works on redis:

```
time     client 1                          client 2
t1         redis-cli
t2         auth user user
t3         multi
t4         set key 23
t5                                              redis-cli
t6                                              setuser kostas -@string
t7         exec --> ERROR, categories changed between `multi/exec`
```

Reply:
t1 to t7 is monotonically increasing

Reply:
I was referring to the case when acl rules change _while_ exec is running

--- 

> BUT validation is done on the parent and not on the stub context

Exactly. When you change a connection from its own thread, you assume no other thread has access to it. So it is safe to mutate variables. BUT with squashing, the parent connection is accessed from multiple threads for validation, so its not possible to mutate it (because `ValidateCommandExecution` might be running on another thread)